cli/run_util.go

```diff
@@ -1066,3 +1093,39 @@ func runWorkflowWithConfiguration(
 
 	return buildRunResults, nil
 }
+
+func initializeStepDir(additionalEnvironments *[]envmanModels.EnvironmentItemModel) (string, error) {
+	// ensure a new testDirPath and if created successfuly then attach it to the step process by and env
+	testDirPath, err := ioutil.TempDir(os.Getenv("BITRISE_TEST_RESULTS_DIR"), "test_result")
+	if err != nil {
+		return "", err
+	}
+	// managed to create the test dir, set the env for it for the next step run
+	*additionalEnvironments = append(*additionalEnvironments, envmanModels.EnvironmentItemModel{
+		"BITRISE_TEST_RESULT_DIR": testDirPath,
+	})
+	return testDirPath, nil
+}
+
+func normalizeTestDir(testDirPath string, testResultStepInfo models.TestResultStepInfo) error {
```

Comment:
could be 2 functions and/or more descriptive name e.g. addTestMetadata

Reply:
these actions are kept together for the logics that the function is intended for